beamer/deploy/commands.py

```diff
@@ -96,3 +97,101 @@ def deploy_base(
     path = artifacts_dir / "base.deployment.json"
     beamer.deploy.artifacts.generate(path, account.address, (resolver,))
     log.info("Generated artifact", path=str(path))
+
+
+@click.command("deploy")
+@click.option(
+    "--rpc-file",
+    "rpc_file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    metavar="PATH",
+    help="Path to the RPC config file.",
+)
+@click.option(
+    "--keystore-file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    metavar="PATH",
+    help="Path to the keystore file.",
+)
+@click.password_option(
+    "--password",
+    type=str,
+    default="",
+    prompt=False,
+    help="The password needed to unlock the keystore file.",
+)
+@click.option(
+    "--artifacts-dir",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    required=True,
+    metavar="DIR",
+    help="The directory to store contract deployment artifacts in.",
+)
+@click.option(
+    "--deploy-mintable-token",
+    is_flag=True,
+    help="Deploy MintableToken.sol on the specified chains.",
+)
+@click.argument(
+    "chain.json",
+    nargs=-1,
+    required=True,
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+)
+def deploy(
+    rpc_file: Path,
+    keystore_file: Path,
+    password: str,
+    artifacts_dir: Path,
+    deploy_mintable_token: bool,
+    **chains: dict,
+) -> None:
+    """Deploy L2 Beamer contracts on the specified chains and set up the trusted call chain."""
+    beamer.util.setup_logging(log_level="DEBUG", log_json=False)
+    _ensure_commit_is_on_remote()
+
+    rpc_info = beamer.deploy.config.load_rpc_info(rpc_file)
+    log.info("Loaded RPC file")
+
+    account = beamer.util.account_from_keyfile(keystore_file, password)
+    log.info("Loaded keyfile", address=account.address)
+
+    path = artifacts_dir / "base.deployment.json"
+    base_deployment = beamer.deploy.artifacts.Deployment.from_file(path)
+    log.info("Loaded base chain deployment", path=str(path), chain_id=base_deployment.chain_id)
+
+    url = rpc_info[base_deployment.chain_id]
+    base_w3 = make_web3(url, account, _margin_gas_price_strategy)
+    assert base_w3.eth.chain_id == base_deployment.chain_id
```

Comment:
> can we somehow add this assert whenever we access the rpc_info?

I'm not sure what you mean... could you please give more details?

Reply:
In this PR we have repeatedly the same pattern

```
chain_id = get_chain_id_from_config_file_or_whereever_from
url = rpc_info[base_deployment.chain_id]
web3 = make_web3(url, account, _margin_gas_price_strategy)
```
In the above code part, we add an assert if the RPC actually corresponds to the same chain_id as defined in the rpc file. But in most other cases we don't have this kind of check. 
When should we assert that this is correct and when not? Is there a reasoning or should we always do the check (I think it would be good).

Reply:
I'm simply not able to see what led you to make this assert here but not in other places or whether it doesn't have a reason why

Reply:
> In this PR we have repeatedly the same pattern
> 
> ```
> chain_id = get_chain_id_from_config_file_or_whereever_from
> url = rpc_info[chain_id]
> web3 = make_web3(url, account, _margin_gas_price_strategy)
> ```
> 
> In the above code part, we add an assert if the RPC actually corresponds to the same chain_id as defined in the rpc file. But in most other cases we don't have this kind of check. When should we assert that this is correct and when not? Is there a reasoning or should we always do the check (I think it would be good).

I agree that it would be good to have these checks and I tried to add them wherever I did `make_web3`. But in this case I missed it. In all other places there is such a check. Thanks for spotting!